File: 3_Semestre/Estrutura_Dados/Aula01/exercicio01.java
package Aula01;

import java.util.Arrays;
import java.util.Random;

public class exercicio01 {

    // Um array unidimensional do tipo inteiro e tamanho igual a 100 deve armazenar
    // valores aleatórios. A saída deve imprimir o maior e o menor valor armazenado.

    public static void main(String[] args) {
        int[] vetor = new int[100];

        gerarValores(vetor, 0, 100);
        imprimirValores(vetor);
    }

    public static void gerarValores(int[] vetor, int min, int max) {
        Random rd = new Random();
        int menor = max;
        int maior = min;

        for (int i = 0; i < vetor.length; i++) {
            vetor[i] = rd.nextInt(max - min) + min;

            menor = (menor < vetor[i]) ? menor : vetor[i];
            maior = (maior > vetor[i]) ? maior : vetor[i];
        }

        System.out.println("menor = " + menor);
        System.out.println("maior = " + maior);
    }

    public static void imprimirValores(int[] vetor) {
        System.out.println(Arrays.toString(vetor));
    }
}

File: 3_Semestre/Estrutura_Dados/Aula01/exercicio01.js
var vetor = new Array(100)

function gerarNum(min, max) {
    min = Math.ceil(min)
    max = Math.floor(max)

    return Math.floor(Math.random() * (max - min) + min)
}

function gerarVetor(vetor, min, max) {
    var menor = max
    var maior = min
    
    for (var i = 0; i < vetor.length; i++) {
        vetor[i] = gerarNum(min, max)

        menor = (menor < vetor[i])? menor : vetor[i]
        maior = (maior > vetor[i])? maior : vetor[i]
    }

    console.log('menor => ' + menor)
    console.log('maior => ' + maior)
}

function imprimirVetor(vetor) {
    console.log(vetor)
}

gerarVetor(vetor, 0, 100)
imprimirVetor(vetor)